From c6a0032e82dc3b1a9f6a68726c4993a87c9570dd Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:35 -0400
Subject: [usb] option: add missing usb_mark_last_busy

Message-id: <1410887463-20674-444-git-send-email-dzickus@redhat.com>
Patchwork-id: 95459
O-Subject: [RHEL7 PATCH 443/591] USB: option: add missing usb_mark_last_busy
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 496969c64a618579f085a87310bc904eb190a71d
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:12 2014 +0200

    USB: option: add missing usb_mark_last_busy

    We should call usb_mark_last_busy in all input paths, including the
    interrupt completion handler.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 2003a66..df91ea9 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -1904,6 +1904,7 @@ static void option_instat_callback(struct urb *urb)
 
  /* Resubmit urb so we continue receiving IRQ data */
  if (status != -ESHUTDOWN && status != -ENOENT) {
+  usb_mark_last_busy(port->serial->dev);
   err = usb_submit_urb(urb, GFP_ATOMIC);
   if (err)
    dev_dbg(dev, "%s: resubmit intr urb failed. (%d)\n",
-- 
1.7.1